internal/k8s/nodes/nodes.go

```diff
@@ -3,9 +3,35 @@
 package nodes
 
 import (
+	"strings"
+
 	corev1 "k8s.io/api/core/v1"
 )
 
+func IsNodeAvailable(n *corev1.Node) (bool, string) {
+	ret := true
+	reason := []string{}
+
+	if IsNetworkUnavailable(n) {
+		ret = ret && false
+		reason = append(reason, "nodeNetworkUnavailable")
+	}
+
+	if IsNodeUnschedulable(n) {
+		ret = ret && false
+		reason = append(reason, "nodeUnschedulable")
+	}
+	return ret, strings.Join(reason, ",")
+}
+
+func IsNodeUnschedulable(n *corev1.Node) bool {
+	if n == nil {
```

Comment:
Is this an expected scenario? If not, I'd have it blow up instead of going defensive.

Reply:
unit test are panicking 

```
 go test ./speaker/...
--- FAIL: TestBGPSpeakerEPSlices (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x198 pc=0x14901e2]

goroutine 67 [running]:
testing.tRunner.func1.2({0x16a7fe0, 0x2748e40})
        /home/kka/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.4.linux-amd64/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
        /home/kka/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.4.linux-amd64/src/testing/testing.go:1634 +0x377
panic({0x16a7fe0?, 0x2748e40?})
        /home/kka/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.4.linux-amd64/src/runtime/panic.go:770 +0x132
go.universe.tf/metallb/internal/k8s/nodes.conditionStatus(...)
```
therefore the pattern that also exists in 
https://github.com/metallb/metallb/blob/main/internal/k8s/nodes/nodes.go#L16-L18
and 
https://github.com/metallb/metallb/blob/main/internal/k8s/nodes/nodes.go#L16-L18

Reply:
fair enough!